src/pb_env.cc

```diff
@@ -233,10 +246,39 @@ EnvironmentManager::ExtractIfNotExtracted(std::string env_path)
         std::string("Failed to get the canonical path for ") + env_path + ".");
   }
 
+  time_t last_modified_time;
+  LastModifiedTime(canonical_env_path, &last_modified_time);
+
+  bool env_extracted = false;
+  bool re_extraction = false;
+  const auto env_itr = env_map_.find(canonical_env_path);
+  if (env_itr != env_map_.end()) {
+    // Check if the environment has been modified and would
+    // need to be extracted again.
+    if (env_itr->second.second == last_modified_time) {
+      env_extracted = true;
+    } else {
+      // Environment file has been updated. Need to clear
+      // the previously extracted environment and extract
+      // the environment to the same destination directory.
+      RecursiveDirectoryDelete(env_itr->second.first.c_str());
+      re_extraction = true;
+    }
+  }
+
   // Extract only if the env has not been extracted yet.
-  if (env_map_.find(canonical_env_path) == env_map_.end()) {
-    std::string dst_env_path(
-        std::string(base_path_) + "/" + std::to_string(env_map_.size()));
+  if (!env_extracted) {
+    LOG_MESSAGE(
+        TRITONSERVER_LOG_VERBOSE,
+        (std::string("Extracting Python execution env ") + canonical_env_path)
```

Comment:
I am using this log to test when the backend performs extraction of the environment.

Reply:
Would it be possible to not rely on the log for testing this bug?

Reply:
I went with the log based approach because it helps get exact visibility into when the re-extraction happens. We can do something like the original issue, but it makes the test much broader and difficult to comprehend. 

Additionally, I think having this extra set of **verbose** logs to look at if the environment was extracted or not will be useful for the user.